doc/kubeletConfig-remediations.md

```diff
@@ -0,0 +1,104 @@
+# KubeletConfig Remediation proposed flow design
+
+This page describes the design and implementation of the KubeletConfig remediation templating support in the compliance operator.
+
+## Goals
+We want to accomplish:
+   * The remediation content can have `KubeletConfig` as payload
+   * We need to update the `parse_arf_result` to be able to take `KubeletConfig`
+   * The remediation controller will be able to merge KubeletConfig remediation into the correct KubeletConfig that is currently being used by the cluster
+
+## Add CaC `KubletConfig` remediation content 
+In CaC content, we are going to add `KubletConfig` remediation besides `MachineConfig` that we currently have.
+
+We will have remdiation that has object payload to be `KubletConfig`. As a exmaple below:
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: KubeletConfig
+metadata:
+  name: co-remediation
+spec:
+  machineConfigPoolSelector:
+    matchLabels:
+      custom-kubelet: compliance-remediaiton
+  kubeletConfig:
+    maxPods: 1100
+```
+
+## Remediation controller handles `KubletConfig` remediaiton
+
+When we are going to apply one of `KubletConfig` remediation the following will happen:
+
+First, we need to get MachineConfigPool list.
+```go
+mcfgpools := &mcfgv1.MachineConfigPoolList{}
+  if err := r.client.List(context.TODO(), mcfgpools); err != nil {
+    return fmt.Errorf("couldn't list the pools for the remediation: %w", err)
+  }
+```
+
+We then find the pool that meets the remediation nodes' label. ex. worker/master nodes `machineconfiguration.openshift.io/role:`
+
+### Remediation when no `KubeletConfigs` present
+
+Once we have the machine config pool, we need to find out if an Admin has created any `KubeletConfigs`, to do that we can check `pool.spec.configuration.source`, if there is only one `name: 01-master-kubelet` machine config in the list.
+
+If there is no `KubeletConfigs` for that pool, we will need to create one, and also add a label to the pool, `custom-kubelet=co-kublet-remediation`.
```

Comment:
Will we have a default naming scheme used by the Compliance Operator for this? If so, we might want to document it here.

Reply:
Yeah, we might want that there! Is there any specific formatted I should follow?

Reply:
we have one for machineconfigs: https://github.com/openshift/compliance-operator/blob/d3ac10754b556dabeff9a107ccb611775d2d28d1/pkg/apis/compliance/v1alpha1/complianceremediation_types.go#L145 which is based on the scan name. You could do something similar. Not requiring the numbering which we use for priority, but the scan name would be a good idea.

Reply:
Agree, that's a good idea.